docs/implementation.adoc

```diff
@@ -0,0 +1,507 @@
+== Implementation Guide
+
+This document provides the guidelines for a correct optimal implementation of the SCM API.
```

Comment:
Yes, i'm monitoring this and would like to make multibranch support and implement non-workflow multibranch, but all this WIP APIs makes difficult understand what is happening.